r/tests/testthat/test-Array.R

```diff
@@ -1397,3 +1397,9 @@ test_that("Can convert R integer/double to decimal (ARROW-11631)", {
     "Conversion to decimal from non-integer/double"
   )
 })
+
+test_that("Array handles sub-day precision Date correctly", {
+  d <- as.Date(-0.1)
+  arr <- arrow_array(d)
+  expect_equal(as.vector(arr), as.Date("1969-12-31"))
+})
```

Comment:
Is it important that the sub-day precision is also a negative here? Would `as.Date(0.1)` exhibit the same issue?

Reply:
Aaah got it